src/mailroom.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+""" This is the mailroom madness script. It will run from the cmd line and
+ allow you to generate an email or report based on user input."""
+
+import os
+import sys
+import math
+
+
+DONOR_DICT = {'Linus Torvalds': [100, 150, 250],
+              'Jane Goodall': [200, 250, 50],
+              'Bob Ross': [1000, 550, 1]}
+
+
+def main_menu_input():
+    '''Display user prompt for main menu'''
+    user_input = input('''
+    Main Menu:
+    E to generate a thank you email to send to a donor
+    R for a report on the donation history of all donors
+    Type exit at any prompt to exit the program.
+
+    >>>''').upper()
+    menu_input_validation(user_input)
+
+
+def email_donor_input():
+    user_input = input('''
+    Please enter in a donor name.
+    Type list for a list of all donors.
+    Type menu to return to the main menu prompt.
+
+    >>>''').upper()
+    email_input_validation(user_input)
+
+
+def email_donation_input():
+    user_input = int(input('''
+    Please enter the donation amount in numbers only.
+    >>>'''))
+    if type(user_input) == int or type(user_input) == float:
+        return float(user_input)
+    else:
+        email_donation_input()
```

Comment:
All three of these functions, if they returned the value the user provided directly instead of calling another function at the end, would be perfect for isolating input.  So close!